src/server/scripts/Outland/CoilfangReservoir/SerpentShrine/boss_lurker_below.cpp

```diff
@@ -59,152 +61,154 @@ const Position positions[MAX_SUMMONS] =
     {42.471519f, -445.115295f, -19.769423f, 0.0f}
 };
 
-class boss_the_lurker_below : public CreatureScript
+struct boss_the_lurker_below : public BossAI
 {
-public:
-    boss_the_lurker_below() : CreatureScript("boss_the_lurker_below") { }
+    boss_the_lurker_below(Creature* creature) : BossAI(creature, DATA_THE_LURKER_BELOW)
+    {
+        scheduler.SetValidator([this]
+        {
+            return !me->HasUnitState(UNIT_STATE_CASTING);
+        });
+    }
 
-    CreatureAI* GetAI(Creature* creature) const override
+    void Reset() override
     {
-        return GetSerpentShrineAI<boss_the_lurker_belowAI>(creature);
+        BossAI::Reset();
+        me->SetReactState(REACT_PASSIVE);
+        me->SetStandState(UNIT_STAND_STATE_SUBMERGED);
+        me->SetVisible(false);
+        me->SetUnitFlag(UNIT_FLAG_NOT_SELECTABLE);
+        // Reset summons
+        //summons.DespawnAll(); shouldn't be needed
     }
 
-    struct boss_the_lurker_belowAI : public BossAI
+    void JustSummoned(Creature* summon) override
     {
-        boss_the_lurker_belowAI(Creature* creature) : BossAI(creature, DATA_THE_LURKER_BELOW) { }
+        summon->SetInCombatWithZone();
+        summons.Summon(summon);
+    }
 
-        void Reset() override
+    void DoAction(int32 param) override
```

Comment:
I see this as both action and param in the core. But I would prefer int32 action